mobile/src/people/people-company.tsx

```diff
@@ -1,23 +1,144 @@
+    departmentIdsBranch?: string[];
 }
 
+interface PeopleCompanyDispatchProps {
+    updateDepartmentIdsTree: (index: number, departmentId: string) => void;
+}
+
+const mapDispatchToProps = (dispatch: Dispatch<PeopleActions>) => ({
+    updateDepartmentIdsTree: (index: number, departmentId: string) => {
+        dispatch(updateDepartmentIdsTree(index, departmentId));
+    },
+});
+
 const mapStateToProps = (state: AppState): PeopleCompanyProps => ({
-    routeName: 'Company'
+    routeName: 'Company',
+    headDepartment: state.organization.departments.filter((department) => department.isHeadDepartment === true)[0],
+    departmentsTree: departmentsTreeFor(state.organization),
+    departmentIdsBranch: state.people.departmentIdsBranch
 });
 
-export class PeopleCompanyImpl extends React.Component<PeopleCompanyProps> {  
+function departmentsTreeFor(organization: OrganizationState) {
+    const topLevelDepartment: Department = organization.departments.filter((department) => department.isHeadDepartment === true)[0];
+    const topLevelEmployee: Employee = organization.employees.employeesById.get(topLevelDepartment.chiefId);
+    const departmentsTree: DepartmentsTree = {
+        root: {
+            departmentId: topLevelDepartment.departmentId,
+            departmentAbbreviation: topLevelDepartment.abbreviation,
+            departmentChiefId: topLevelDepartment.chiefId,
+            head: topLevelEmployee,
+            parent: null,
+            children: childrenNodes(topLevelDepartment, organization.departments, organization.employees.employeesById),
+            subordinates: organization.employees.employeesById.filter((employee) => employee.departmentId === topLevelDepartment.departmentId).toArray()
+        }
+    };
+
+    return departmentsTree;
+}
+
+function childrenNodes(headDeaprtment: Department, departments: Department[], employees: EmployeeMap) {
+    var nodes: DepartmentsTreeNode[] = [];
+    const sublings = departments.filter((subDepartment) => subDepartment.parentDepartmentId === headDeaprtment.departmentId);
+    sublings.forEach(department => {
+        const employee: Employee = employees.get(department.chiefId);
+        const subSublings = departments.filter((subDepartment) => subDepartment.parentDepartmentId === department.departmentId);
+        nodes.push({
+            departmentId: department.departmentId,
+            departmentAbbreviation: department.abbreviation,
+            departmentChiefId: department.chiefId,
+            head: employee,
+            parent: headDeaprtment,
+            children: subSublings.length > 0 ? childrenNodes(department, departments, employees) : null,
+            subordinates: employees.filter((emp) => emp.departmentId === department.departmentId).toArray()
+        });
+    });
+
+    return nodes;
+}
+
+export class PeopleCompanyImpl extends React.Component<PeopleCompanyProps & PeopleCompanyDispatchProps> {
+    public componentWillMount() {
```

Comment:
Don't use componentWillMount.

https://github.com/reactjs/rfcs/blob/master/text/0006-static-lifecycle-methods.md

It should be move to init state with logic above